File: lib/jets/builders/templates/handler.js
'use strict';

const shim = require("handlers/shim.js");

shim.once(); // runs in lambda execution context

<% @vars.functions.each do |function_name| -%>
exports.<%= function_name %> = shim.handler("<%= @vars.handler_for(function_name) %>");
<% end -%>

File: lib/jets/builders/templates/handler.rb
require "bundler/setup"
require "jets"
Jets.once  # runs once in lambda execution context

<% @vars.functions.each do |function_name|
  handler = @vars.handler_for(function_name)
  meth = handler.split('.').last
  -%>
def <%= meth -%>(event:, context:)
  ActiveRecord::Base.connection.reconnect! unless ActiveRecord::Base.connection.active?
  Jets.process(event, context, "<%= handler -%>")
ensure
  ActiveRecord::Base.connection.disconnect!
end
<% end %>
